### PR TITLE
User registration added 

### PR DESCRIPTION
**Pull request by mike2151**
Added user registration with a simple signup UI.

### CODE REVIEW

File: apps/reader/models.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
 from django.db import models
 
-from django.contrib.auth.models import User
+#from django.contrib.auth.models import User
```
Comment (jreinstra) [inline apps/reader/models.py]: remove comment

File: apps/reader/models.py [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
 from apps.director.models import Hackathon, Organization
 from apps.registration.models import Application, Applicant
 
+from django.contrib.auth import get_user_model
 
 class Reader(models.Model):
+    User = get_user_model()
```
Comment (jreinstra) [inline apps/reader/models.py]: why are you using this?

File: apps/registration/apps.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
+
+
+class BlogConfig(AppConfig):
```
Comment (jreinstra) [inline apps/registration/apps.py]: rename class

File: apps/registration/forms.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+from django import forms
+from django.contrib.auth.forms import UserCreationForm
+#from django.contrib.auth.models import User
+from .models import Applicant
+
+from django.contrib.auth import get_user_model
+
+
+class SignupForm(UserCreationForm):
```
Comment (jreinstra) [inline apps/registration/forms.py]: I'd prefer is we make this an API endpoint rather than Django form

Comment (jreinstra) [reply apps/registration/forms.py]: is that workable for you?

File: apps/registration/migrations/0006_auto_20180125_2313.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.5 on 2018-01-26 04:13
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (jreinstra) [inline apps/registration/migrations/0006_auto_20180125_2313.py]: I think you can probably just nuke the database and restart the models from scratch - no need to create unnecessary model things and then remove them in a migration - that'll change once we have a live system

File: apps/registration/models.py [hunk 6]
```diff
@@ -17,52 +15,21 @@ class Application(models.Model):
 
     name = models.CharField(max_length=128)
     status = models.CharField(max_length=16, choices=STATUS_CHOICES)
-    hackathon = models.ForeignKey(Hackathon, related_name="applications")
+    #hackathon = models.ForeignKey(Hackathon, related_name="applications")
```
Comment (jreinstra) [inline apps/registration/models.py]: remove comment

File: apps/registration/models.py [hunk 7]
```diff
@@ -17,52 +15,21 @@ class Application(models.Model):
...
-
-
-class ApplicantTeam(models.Model):
-    hackathon = models.ForeignKey(Hackathon)
-    name = models.CharField(max_length=64)
-    entry_code = models.CharField(max_length=64)
-
-    def __str__(self):
-        return "{} ({})".format(self.name, self.hackathon)
+# Create your models here.
```
Comment (jreinstra) [inline apps/registration/models.py]: remove comment

File: apps/registration/views.py [hunk 8]
```diff
@@ -1,39 +1,58 @@
...
+            user.is_active = False
+            user.save()
+            current_site = get_current_site(request)
+            message = render_to_string('acc_active_email.html', {
+                'user':user, 'domain':current_site.domain,
+                'uid': urlsafe_base64_encode(force_bytes(user.pk)),
+                'token': account_activation_token.make_token(user),
+            })
+            # Sending activation link in terminal
+            # user.email_user(subject, message)
```
Comment (jreinstra) [inline apps/registration/views.py]: remove comments

File: madras/settings.py [hunk 9]
```diff
@@ -83,13 +85,36 @@
 WSGI_APPLICATION = 'madras.wsgi.application'
 
 
+# EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
```
Comment (jreinstra) [inline madras/settings.py]: rm

File: apps/reader/models.py (hunk 2 above)
Comment (mike2151) [reply apps/reader/models.py]: Because I altered the user model

**Pull request closed without merging**